File: trunk/ShadowFrameworkLite20/src/shadow/geometry/geometries/structures/SFQuadsGrid.java
package shadow.geometry.geometries.structures;


public class SFQuadsGrid {
	
	private boolean closeU;
	private boolean closeV;
	private float[] vSplits;
	private float[] uSplits;

	public SFQuadsGrid() {
	}

	public boolean isCloseU() {
		return closeU;
	}

	public void setCloseU(boolean closeU) {
		this.closeU = closeU;
	}

	public boolean isCloseV() {
		return closeV;
	}

	public void setCloseV(boolean closeV) {
		this.closeV = closeV;
	}

	public float[] getvSplits() {
		return vSplits;
	}

	public void setvSplits(float[] vSplits) {
		this.vSplits = vSplits;
	}

	public float[] getuSplits() {
		return uSplits;
	}

	public void setuSplits(float[] uSplits) {
		this.uSplits = uSplits;
	}

	public float[] getU_splits() {
		return getuSplits();
	}

	public void setU_splits( float[] uSplits) {
		setuSplits(uSplits);
	}

	public float[] getV_splits() {
		return getvSplits();
	}

	public void setV_splits(float[] vSplits) {
		setvSplits(vSplits);
	}

	public int getNu() {
		return getuSplits().length;
	}

	public void setNu( int nu) {
		setU_splits(new float[nu]);
		float stepU=1.0f/(nu-1.0f);
		for (int i = 0; i < nu; i++) {
			getU_splits()[i]=i*stepU;
		}
	}

	public int getNv() {
		return getvSplits().length;
	}

	public void setNv(int nv) {
		setV_splits(new float[nv]);
		float stepV=1.0f/(nv-1.0f);
		for (int i = 0; i < nv; i++) {
			getV_splits()[i]=i*stepV;
		}
	}
	
	public static int getPartitionedSplitsSize(int n1,int splitsCount){
		return (splitsCount-1)*n1+1;
	}

	public float[] generatePartitionedSplits( int n1, float stepn1, float[] vSplits) {
		float[] vs=new float[(vSplits.length-1)*n1+1];
		for(int i=0;i<(vSplits.length-1)*n1+1;i++){
			int I=i/n1;
			int Ires=i-n1*(I);
			vs[i]=vSplits[I]+(Ires>0?Ires*stepn1*(vSplits[I+1]-vSplits[I]):0);
		}
		return vs;
	}
}

File: trunk/ShadowFrameworkJS/src/shadow/geometry/geometries/structures/SFQuadsGrid.js




function SFQuadsGrid(){
	this.closeU=false;
	this.closeV=false;
}

SFQuadsGrid.prototype["isCloseU"]=function(){
	return this.closeU;
};

SFQuadsGrid.prototype["isCloseV"]=function(){
	return this.closeV;
};

SFQuadsGrid.prototype["getvSplits"]=function(){
	return this.vSplits;
};

SFQuadsGrid.prototype["getuSplits"]=function(){
	return this.uSplits;
};


SFQuadsGrid.prototype["setCloseU"]=function(closeU){
	this.closeU=closeU;
};

SFQuadsGrid.prototype["setCloseV"]=function(closeV){
	this.closeV=closeV;
};

SFQuadsGrid.prototype["setvSplits"]=function(vSplits){
	this.vSplits=vSplits;
};

SFQuadsGrid.prototype["setuSplits"]=function(uSplits){
	this.uSplits=uSplits;
};

SFQuadsGrid.prototype["getU_splits"]=function(){
	return this.getuSplits();
};

SFQuadsGrid.prototype["getV_splits"]=function(){
	return this.getvSplits();
};

SFQuadsGrid.prototype["setU_splits"]=function(uSplits){
	this.setuSplits(uSplits);
};

SFQuadsGrid.prototype["setV_splits"]=function(vSplits){
	this.setvSplits(vSplits);
};


SFQuadsGrid.prototype["getNu"]=function(){
	return this.getuSplits().length;
};


SFQuadsGrid.prototype["setNu"]=function(nu){
	this.setU_splits(new Array());
	var stepU=1.0/(nu-1.0);
	for (var i = 0; i < nu; i++) {
		this.uSplits[i]=i*stepU;
	}
};


SFQuadsGrid.prototype["getNv"]=function(){
	return this.getvSplits().length;
};

	

SFQuadsGrid.prototype["setNv"]=function(nv){
	this.setV_splits(new Array());
	var stepV=1.0/(nv-1.0);
	for (var i = 0; i < nv; i++) {
		this.vSplits[i]=i*stepV;
	}
};

function SFQuadsGrid_getPartitionedSplitsSize(n1,splitsCount){
	return (splitsCount-1)*n1+1;
}

SFQuadsGrid.prototype["generatePartitionedSplits"]=function(n1,stepn1,vSplits){
	var size=(vSplits.length-1)*n1+1;
	var vs=new Array();
	for(var i=0;i<size;i++){
		var I=Math.floor(i/n1);
		var Ires=i-n1*(I);
		vs[i]=vSplits[I]+(Ires>0?Ires*stepn1*(vSplits[I+1]-vSplits[I]):0);
		
	}
	return vs;
};


